src/async.rs

```diff
@@ -79,6 +79,30 @@ impl AsyncClient {
         }
     }
 
+    /// Get a [`Txid`] of a transaction given its index in a block with a given hash.
+    pub async fn get_txid_at_block_index(
+        &self,
+        block_hash: &BlockHash,
+        index: usize,
+    ) -> Result<Option<Txid>, Error> {
+        let resp = self
+            .client
+            .get(&format!(
+                "{}/block/{}/txid/{}",
+                self.url,
+                block_hash.to_string(),
+                index
+            ))
+            .send()
+            .await?;
+
+        if let StatusCode::NOT_FOUND = resp.status() {
+            return Ok(None);
+        }
+
+        Ok(Some(deserialize(&Vec::from_hex(&resp.text().await?)?)?))
```

Comment:
I think you can simplify this by using `Txid::from_str()`: https://docs.rs/bitcoin/0.29.1/bitcoin/hash_types/struct.Txid.html#impl-FromStr-for-Txid

Reply:
Ah, good point! Fixed with https://github.com/bitcoindevkit/rust-esplora-client/pull/8/commits/24ccf9cacd5d54f933f7c2a10bc2275beae0a22e.